### PR TITLE
Jira OSD-5709: Link RouteMonitor CR to ServiceMonitor CR

### PR DESCRIPTION
**Pull request by NautiluX**
Creating a new PR since @rafael-azevedo is out.

This PR fixes the integration test and speeds up rebuilding docker images, which happens on each integration-test run

### CODE REVIEW

**Comment by RiRa12621** [comment]
Closing https://github.com/openshift/route-monitor-operator/pull/50 as this is replacing it

**Comment by RiRa12621** [comment]
/retitle Jira OSD-5709: Link RouteMonitor CR to ServiceMonitor CR

**Review by georgettica — COMMENTED** [review]
aside of that 
/lgtm
putting on
/hold
in case you want to address the comment before it's merged

File: api/v1alpha1/common_types.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package v1alpha1
+
+// NamespacedName contains the name of a object and its namespace
+type NamespacedName struct {
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply api/v1alpha1/common_types.go:4]: There is [this type](https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go#L27) already, but it doesn't have the json pieces, that's way each operator has to redefine it.

**Pull request merged**